src/post/api/getPosts.ts

```diff
@@ -1,9 +1,23 @@
 import { Post } from "../types";
 
-const getPosts = async (): Promise<Post> => {
+const getPosts = async (): Promise<Post[]> => {
   const apiResponse = await fetch(`${import.meta.env.VITE_API_URL}posts`);
 
-  return apiResponse.json();
+  const apiPostsDto = (await apiResponse.json()) as Post[];
+
+  const posts = apiPostsDto.map<Post>((apiPostDto) => {
+    const post: Post = {
+      id: apiPostDto.id,
```

Comment:
En esta conversión estáis totalmente acoplados a todas las propiedades del objeto, y sólo deberíais acoplaros a una, como hice en mi demo.

Reply:
Cierto, tendría que haber utilizado el operador spread.